programs/clearing_house/src/controller/pnl.rs

```diff
@@ -124,3 +137,114 @@ pub fn settle_pnl(
 
     Ok(())
 }
+
+pub fn settle_expired_position(
+    market_index: u64,
+    user: &mut User,
+    user_key: &Pubkey,
+    market_map: &MarketMap,
+    bank_map: &BankMap,
+    oracle_map: &mut OracleMap,
+    now: i64,
+    fee_structure: &FeeStructure,
+) -> ClearingHouseResult {
+    {
+        let bank = &mut bank_map.get_quote_asset_bank_mut()?;
+        update_bank_cumulative_interest(bank, now)?;
+    }
+
+    settle_funding_payment(
+        user,
+        user_key,
+        market_map.get_ref(&market_index)?.deref(),
+        now,
+    )?;
+
+    // cannot settle pnl this way on a user who is in liquidation territory
+    if !(meets_maintenance_margin_requirement(user, market_map, bank_map, oracle_map)?) {
+        return Err(ErrorCode::InsufficientCollateralForSettlingPNL);
+    }
+
+    let position_index = get_position_index(&user.positions, market_index)?;
+
+    let bank = &mut bank_map.get_quote_asset_bank_mut()?;
+    let market = &mut market_map.get_ref_mut(&market_index)?;
+    validate!(
+        market.status == MarketStatus::Settlement,
+        ErrorCode::DefaultError,
+        "Market isn't in settlement"
+    )?;
+
+    let _oracle_price = oracle_map.get_price_data(&market.amm.oracle)?.price;
+    let (base_asset_value, unrealized_pnl) = calculate_base_asset_value_and_pnl_with_oracle_price(
+        &user.positions[position_index],
+        market.settlement_price,
+    )?;
+
+    let fee = base_asset_value
+        .checked_mul(fee_structure.fee_numerator)
+        .ok_or_else(math_error!())?
+        .checked_div(fee_structure.fee_denominator)
+        .ok_or_else(math_error!())?;
+
+    let unrealized_pnl_with_fee = unrealized_pnl
+        .checked_sub(cast_to_i128(fee)?)
+        .ok_or_else(math_error!())?;
+
+    let pnl_to_settle_with_user = update_pnl_pool_balance(market, bank, unrealized_pnl_with_fee)?;
```

Comment:
i think itd be nice to have the update bank for the pnl pool and for the user balance more closely together. makes it harder to read with part of it happening in `update_pnl_pool_balance` and then other in this function. id either put both of em in this function or take them both out as it makes it more legible